app/src/main/java/com/android/voyageur/ui/trip/activities/ActivitiesForOneDay.kt

```diff
@@ -46,36 +55,81 @@ fun ActivitiesForOneDayScreen(
       tripsViewModel.selectedDay.value
           ?: return Text(text = "No day selected. Should not happen", color = Color.Red)
 
+  // States for filtering and search
+  var selectedFilters by remember { mutableStateOf(setOf<ActivityType>()) }
+  var searchQuery by remember { mutableStateOf("") }
+
   var activities by remember {
     mutableStateOf(
         tripsViewModel
             .getActivitiesForSelectedTrip()
             .filter {
-              it.startTime.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate() == day
+              val matchesDate =
+                  it.startTime.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate() ==
+                      day
+              val matchesSearch =
+                  searchQuery.isEmpty() || it.title.contains(searchQuery, ignoreCase = true)
+              val matchesFilter = selectedFilters.isEmpty() || it.activityType in selectedFilters
+              matchesDate && matchesSearch && matchesFilter
+            }
+            .sortedWith(compareBy({ it.startTime }, { it.endTime })))
+  }
+
+  var showDialog by remember { mutableStateOf(false) }
+  var activityToDelete by remember { mutableStateOf<Activity?>(null) }
+  var pricePerDay by remember { mutableStateOf(0.0) }
+
+  // Update activities when search query or filters change
+  LaunchedEffect(searchQuery, selectedFilters) {
+    activities =
+        tripsViewModel
+            .getActivitiesForSelectedTrip()
+            .filter {
+              val matchesDate =
+                  it.startTime.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate() ==
+                      day
```

Comment:
You are reusing this part of the code multiple times. Could you maybe put it in another function such that we reduce duplicated code?